### PR TITLE
Optimized likelihood

### PR DESCRIPTION
**Pull request by laurenzkeller**
I removed the parallel code because it didn't lead to a speed up. Should I add a few comments to the _backend_code.py file such that you understand what I did? Or should I do this later, when we have a draft of the report?

### CODE REVIEW

File: src/pmhn/_trees/_backend_code.py [hunk 1]
```diff
@@ -135,38 +45,75 @@ def loglikelihood(
...
+                            V_new[index] += lamb
+                            genotype_list[subclone_index] = 0
+            V_new[index] += sampling_rate
+
+            x = -V_old[index] / V_new[index]
+            if index == subtrees_size - 1:
+                return np.log(x + self._jitter) + np.log(sampling_rate)
+            V_old += V_new * x
+            V_new = np.zeros_like(V_new)
+            x = 0.0
```
Comment (pawel-czyz) [inline src/pmhn/_trees/_backend_code.py]: Wouldn't it be clearer to initialize `x = 0.0` at the beginning of each iteration? Currently it's initialized before the loop starts and at the end of each iteration is zeroed again. (I could have overlooked something, though.)

**Review by pawel-czyz — APPROVED** [review]
Looks good to me!

File: src/pmhn/_trees/_backend_code.py (hunk 1 above)
Comment (laurenzkeller) [reply src/pmhn/_trees/_backend_code.py]: You're totally right, I'll change it.

**Comment by pawel-czyz** [comment]
> Should I add a few comments to the _backend_code.py file such that you understand what I did?

It'll be useful, but I'd suggest to indeed prioritize the report now – we'll go back to documenting the code later :slightly_smiling_face:

**Pull request merged**